pytorch_lightning/strategies/single_device.py

```diff
@@ -20,20 +22,21 @@
 from pytorch_lightning.plugins.precision import PrecisionPlugin
 from pytorch_lightning.strategies.strategy import Strategy
 from pytorch_lightning.utilities import _XLA_AVAILABLE
+from pytorch_lightning.utilities.types import _DEVICE
 
 
 class SingleDeviceStrategy(Strategy):
     """Strategy that handles communication on a single device."""
 
     def __init__(
         self,
-        device: torch.device,
-        accelerator: Optional["pl.accelerators.accelerator.Accelerator"] = None,
-        checkpoint_io: Optional[CheckpointIO] = None,
-        precision_plugin: Optional[PrecisionPlugin] = None,
+        device: _DEVICE,
+        accelerator: pl.accelerators.accelerator.Accelerator | None = None,
+        checkpoint_io: CheckpointIO | None = None,
```

Comment:
quick ques: is this new API supported on 3.6? since we test that on the release branch.

Reply:
@rohitgr7 I don't think it's supported on 3.6 as PEP 585 which is one of the PEPs we're addressing mentions: https://www.python.org/dev/peps/pep-0585/#backwards-compatibility
> Python files with the "annotations" future-import (available since Python 3.7) can parameterize...

---

@carmocca #11117 should be cherry picked into the weekly release branch, right? since Python 3.6 ended its life.

Reply:
@akihironitta we plan to still support it during all `1.5.x` releases